docs: add data provider consent spec for protocol devs (#592)
medibloc/panacea-core: .gitbook/3-protocol-devs/1-dep-specs/3-data-provider-consent.md

# Data Provider Consent

- Status: Draft
- Crated: 2023-01-04
- Modified: 2023-01-04
- Authors
    - Youngjoon Lee <yjlee@medibloc.org>
    - Gyuguen Jang <gyuguen.jang@medibloc.org>
    - Hansol Lee <hansol@medibloc.org>
    - Myongsik Gong <myongsik_gong@medibloc.org>
    - Inchul Song <icsong@medibloc.org>

## Synopsis

This document is about providing data through [DEP](../../1-users/3-data-exchange/0-about-dep.md).
Before providers provide their data to consumers, the data must be validated from oracle whether the data is the correct data specified in the deal.
Providers can send a request for validation to one of the oracles registered in panacea, and the oracle will validate the data using confidential computing without any exposure of the data.

> For the data that has been validated, oracle will issue a certificate to the provider.

Reviewer: ```suggestion
As a result of a data validation, oracle will issue a certificate to the provider.
```

- `Deal` is defined in [Data Deal](./2-data-deal.md)
- `Certificate`: a certificate that the data is validated to be provided to the deal, which is issued by oracle.

## Technical Specification

Before provider provides the data to deal, data should be validated by the oracle.
If the data is successfully validated to be provided, the provider will have a `Certificate` like below (more about [Data Validation](./4-data-validation.md)):

```proto
message Certificate {
  UnsignedCertificate unsigned_certificate = 1;
  bytes signature = 2;
}

message UnsignedCertificate {
    string cid = 1;
    string unique_id = 2;
    string oracle_address = 3;
    uint64 deal_id = 4;
    string provider_address = 5;
    string data_hash = 6;
}
```

Using the `Certificate`, provider can submit consent to provide the data.

```proto
message MsgSubmitConsent {
  Consent consent = 1;
}

message Consent {
  Certificate certificate = 1;
}
```

When this consent is submitted, blockchain will check:
- if the data is provided by the owner of the data
- if the data is validated by a registered and active oracle
- if the data is provided in duplicate

> If all checks pass, rewards are distributed to the provider and oracle(more about [incentive](./6-incentives.md)).

Reviewer 2: ```suggestion
If all checks are passed, rewards are distributed to the provider and oracle(more about [incentive](./6-incentives.md)).
```

Author: I used the `pass` as an intransitive verb which have no direct object.

Reviewer 2: Ah, I'm not sure my comment is right. My purpose is change to past verb.
